meshtastic/node_runtime/transport_runtime.py

```
@@ -0,0 +1,490 @@
+            Longitude value. Floats are interpreted as degrees and scaled by
+            1e7; ints are treated as pre-scaled ``longitude_i`` values.
+            ``None`` and zero values (``0``/``0.0``) omit ``longitude_i`` from
+            the sent message for backward compatibility with sentinel semantics.
+        alt : int | None
+            Altitude in meters. ``None`` omits altitude from the sent message.
+        """
+        self._node.ensureSessionKey()
+
+        # Type validation: reject bool and non-numeric types
+        if lat is not None:
+            if isinstance(lat, bool) or not isinstance(lat, (int, float)):
```

Ruff (0.15.6)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)